driver-core/src/test/java/com/datastax/driver/core/DirectedGraphTest.java

```diff
@@ -58,24 +68,80 @@ public void should_sort_complex_graph() {
         g.addEdge("D", "B");
         g.addEdge("B", "A");
 
-        // Topological sort order should be : GH,FE,D,CB,A
+        // Topological sort order should be : GH,E,F,D,BC,A
         // There's no guarantee on the order within the same level, so we use sublists:
         List<String> sorted = g.topologicalSort();
+        System.out.println(sorted);
+        assertThat(sorted.subList(0, 2))
+                .containsExactly("G", "H");
+        assertThat(sorted.subList(2, 3))
+                .containsExactly("E");
+        assertThat(sorted.subList(3, 4))
+                .containsExactly("F");
+        assertThat(sorted.subList(4, 5))
+                .containsExactly("D");
+        assertThat(sorted.subList(5, 7))
+                .containsExactly("B", "C");
+        assertThat(sorted.subList(7, 8))
+                .containsExactly("A");
+    }
+
+    @Test(groups = "unit")
+    public void should_sort_complex_custom_comparator() {
+        // Version of should_sort_complex_graph using a custom comparator based on ordering largest values first.
+        // This is counter to how hashmaps should usually behave, so this should help ensure that the comparator is
+        // being used.
+        Comparator<Integer> highFirst = new Comparator<Integer>() {
+            @Override
+            public int compare(Integer o1, Integer o2) {
+                return o2 - o1;
+            }
+        };
+
+        // sort graph and use a alphaComparator that favors larger values ordered first.
+        //         7   6
+        //        / \ /\
+        //       5   | 10
+        //        \ /  /
+        //         9  /
+        //        / \/
+        //        1  2
+        //        |
+        //        0
+        DirectedGraph<Integer> g = new DirectedGraph<Integer>(highFirst, 0, 1, 2, 9, 10, 5, 6, 7);
+        g.addEdge(7, 5);
+        g.addEdge(6, 10);
+        g.addEdge(7, 9);
+        g.addEdge(5, 9);
+        g.addEdge(6, 9);
+        g.addEdge(9, 2);
+        g.addEdge(10, 2);
+        g.addEdge(9, 1);
+        g.addEdge(1, 0);
+
+        // Topological sort order should be : [7,6],[5],[10],[2,1],[0]
+        // There's no guarantee on the order within the same level, so we use sublists:
+        List<Integer> sorted = g.topologicalSort();
+        System.out.println(sorted);
         assertThat(sorted.subList(0, 2))
-                .contains("G", "H");
-        assertThat(sorted.subList(2, 4))
-                .contains("F", "E");
+                .containsExactly(7, 6);
+        // 5 comes before 10 even though they appear at the same depth.  This happens because 5's (7) dependency
```

Comment:
Note this detail, since 7's dependents (5,9) are evaluated first, 5 comes before 10, even though they are the same depth.   I don't expect this to bother anyone, but I thought I'd expound on that detail.